pyopae/README.md

```diff
@@ -0,0 +1,123 @@
+|----------------|-----------------|----------------|--------------|
+| 2.7 | CPython w/ UCS4 | x86_64 | opae.fpga.@INTEL_FPGA_API_VERSION@-cp27-cp27mu-linux_x86_64.whl |
+| 3.4 | CPython w/ UCS4 | x86_64 | opae.fpga.@INTEL_FPGA_API_VERSION@-cp34-cp34mu-linux_x86_64.whl |
+| 3.6 | CPython w/ UCS4 | x86_64 | opae.fpga.@INTEL_FPGA_API_VERSION@-cp36-cp36mu-linux_x86_64.whl |
+
+
+opae.fpga is currently not available in the Python Package Index but once it
+does become available, one should be able to install using pip by simply typing
+the following:
+```shell
+> pip install --user opae.fpga
+```
+
+## Intalling From Source
+In addition to the runtime libraries mentioned above, installing from source
+does require that the OPAE header files be installed as well as those header
+files for pybind11. The former can be installed with the opae-devel package and
+the latter can be installed by installing pybind11 Python module.
+
+### Example Installation
+The following example shows how to build from source by installing the
+prerequisites before running the setup.py file.
+
+```shell
+>sudo yum install opae-libs-@INTEL_FPGA_API_VERSION@-@INTEL_FPGA_API_RELEASE@.x86_64.rpm
+>sudo yum install opae-devel-@INTEL_FPGA_API_VERSION@-@INTEL_FPGA_API_RELEASE@.x86_64.rpm
+>pip install --user pybind11
+>pip install --user opae.fpga-@INTEL_FPGA_API_VERSION@.tar.gz
+```
+
+
+_NOTE_: The `pip` examples above use the `--user` flag to avoid requiring root
+permissions. Those packages will be installed in the user's `site-packages`
+directory found in the user's `.local` directory.
+
+## Example Scripts
+The following example is an implementation of the sample, hello_fpga.c, which
+is designed to configure the NLB0 diagnostic accelerator for a simple loopback.
+
+
+```Python
+import time
+from opae import fpga
+
+NLB0 = "d8424dc4-a4a3-c413-f89e-433683f9040b"
+CTL = 0x138
+CFG = 0x140
+NUM_LINES = 0x130
+SRC_ADDR = 0x0120
+DST_ADDR = 0x0128
+DSM_ADDR = 0x0110
+
+def cl_align(addr):
+    return addr >> 6
+
+tokens = fpga.enumerate(type=fpga.ACCELERATOR, guid=NLB0)
+assert tokens, "Could not enumerate accelerator: {}".format(NlB0)
+
+with fpga.open(tokens[0], fpga.OPEN_SHARED) as handle:
+    src = fpga.allocate_shared_buffer(handle, 4096)
+    dst = fpga.allocate_shared_buffer(handle, 4096)
+    dsm = fpga.allocate_shared_buffer(handle, 4096)
+    handle.write_csr32(CTL, 0)
+    handle.write_csr32(CTL, 1)
+    handle.write_csr64(DSM_ADDR, dsm.io_address())
+    handle.write_csr64(SRC_ADDR, cl_align(src.io_address())) # cacheline-aligned
+    handle.write_csr64(DST_ADDR, cl_align(dst.io_address())) # cacheline-aligned
+    handle.write_csr32(CFG, 0x42000)
+    handle.write_csr32(NUM_LINES, 4096/64)
+    handle.write_csr32(CTL, 3)
+    while dsm[0x42] & 0x1 == 0:
+        time.sleep(0.001)
+    handle.write_csr32(CTL, 7)
+
+```
+
+This example shows how one might reprogram (Partial Reconfiguration) an
+accelerator on a given bus, 0x5e, using a bitstream file, m0.gbs.
```

Comment:
You mention PCI bus 0x5e here, but I don't see it used in the example code.  That's a bit confusing.